Developing.md

```diff
@@ -1,7 +1,7 @@
 # Getting Started with Developing CUDA Quantum
 
 This document contains guidelines for contributing to the code in this
-repository. This document is relevant primarily for contributions to the NVQ++
+repository. This document is relevant primarily for contributions to the `nvq++`
```

Comment:
Probably could split the documentation updates into a separate PR. (This one is really long.)

Reply:
Yes, the spell checker update go bigger than expected - I didn't per se plan that but was annoyed enough with it to want to set it up properly. I can pull the repo checks changes into their own PR.